ESP32 core 2.0.3 fails to compile with errors and warning

Building for an ESP32 board espressif core 2.0.3 will give the following errors:
```
In file included from /Users/user/Documents/Arduino/libraries/souliss/Souliss.h:222,
                 from /Users/user/Documents/e01_Hello_ESP32_test/e01_Hello_ESP32_test.ino:30:
/Users/user/Documents/Arduino/libraries/souliss/base/LocalIO.cpp: In function 'U8 Souliss_DigInHold_Helper(U8, U8, U8, U8, U8*, U8, U16)':
/Users/user/Documents/Arduino/libraries/souliss/base/LocalIO.cpp:299:49: error: call of overloaded 'abs(long unsigned int)' is ambiguous
  else if(pin_value && (abs(millis()-souliss_time) > holdtime) && (InPin[pin]==PINSET))
                                                 ^
In file included from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/cstdlib:75,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/stdlib.h:36,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/assert.h:21,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/sys/reent.h:503,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/sys/reent.h:17,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdio.h:60,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/cores/esp32/Arduino.h:27,
                 from sketch/e01_Hello_ESP32_test.ino.cpp:1:
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdlib.h:74:5: note: candidate: 'int abs(int)'
 int abs (int);
     ^~~
In file included from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/cstdlib:77,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/stdlib.h:36,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/assert.h:21,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/sys/reent.h:503,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/sys/reent.h:17,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdio.h:60,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/cores/esp32/Arduino.h:27,
                 from sketch/e01_Hello_ESP32_test.ino.cpp:1:
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:78:3: note: candidate: 'constexpr long double std::abs(long double)'
   abs(long double __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:74:3: note: candidate: 'constexpr float std::abs(float)'
   abs(float __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:70:3: note: candidate: 'constexpr double std::abs(double)'
   abs(double __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:61:3: note: candidate: 'long long int std::abs(long long int)'
   abs(long long __x) { return __builtin_llabs (__x); }
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:56:3: note: candidate: 'long int std::abs(long int)'
   abs(long __i) { return __builtin_labs(__i); }
   ^~~
In file included from /Users/user/Documents/Arduino/libraries/souliss/Souliss.h:222,
                 from /Users/user/Documents/e01_Hello_ESP32_test/e01_Hello_ESP32_test.ino:30:
/Users/user/Documents/Arduino/libraries/souliss/base/LocalIO.cpp: In function 'U8 Souliss_DigInHold2_Helper(U8, U8, U8, U8, U8*, U8, U8, U16)':
/Users/user/Documents/Arduino/libraries/souliss/base/LocalIO.cpp:343:49: error: call of overloaded 'abs(long unsigned int)' is ambiguous
  else if(pin_value && (abs(millis()-souliss_time) > holdtime) && (InPin[pin]==PINSET))
                                                 ^
In file included from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/cstdlib:75,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/stdlib.h:36,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/assert.h:21,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/sys/reent.h:503,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/sys/reent.h:17,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdio.h:60,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/cores/esp32/Arduino.h:27,
                 from sketch/e01_Hello_ESP32_test.ino.cpp:1:
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdlib.h:74:5: note: candidate: 'int abs(int)'
 int abs (int);
     ^~~
In file included from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/cstdlib:77,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/stdlib.h:36,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/assert.h:21,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/sys/reent.h:503,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/sys/reent.h:17,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdio.h:60,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/cores/esp32/Arduino.h:27,
                 from sketch/e01_Hello_ESP32_test.ino.cpp:1:
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:78:3: note: candidate: 'constexpr long double std::abs(long double)'
   abs(long double __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:74:3: note: candidate: 'constexpr float std::abs(float)'
   abs(float __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:70:3: note: candidate: 'constexpr double std::abs(double)'
   abs(double __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:61:3: note: candidate: 'long long int std::abs(long long int)'
   abs(long long __x) { return __builtin_llabs (__x); }
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:56:3: note: candidate: 'long int std::abs(long int)'
   abs(long __i) { return __builtin_labs(__i); }
   ^~~
In file included from /Users/user/Documents/Arduino/libraries/souliss/Souliss.h:222,
                 from /Users/user/Documents/e01_Hello_ESP32_test/e01_Hello_ESP32_test.ino:30:
/Users/user/Documents/Arduino/libraries/souliss/base/LocalIO.cpp: In function 'U8 Souliss_DigKeepHold_Helper(U8, U8, U8, U8, U8*, U8, U16)':
/Users/user/Documents/Arduino/libraries/souliss/base/LocalIO.cpp:398:49: error: call of overloaded 'abs(long unsigned int)' is ambiguous
  else if(pin_value && (abs(millis()-souliss_time) > holdtime) && ((InPin[pin]==PINSET) || (InPin[pin]==PINACTIVE)))
                                                 ^
In file included from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/cstdlib:75,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/stdlib.h:36,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/assert.h:21,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/sys/reent.h:503,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/sys/reent.h:17,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdio.h:60,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/cores/esp32/Arduino.h:27,
                 from sketch/e01_Hello_ESP32_test.ino.cpp:1:
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdlib.h:74:5: note: candidate: 'int abs(int)'
 int abs (int);
     ^~~
In file included from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/cstdlib:77,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/stdlib.h:36,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/assert.h:21,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/sys/reent.h:503,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/sys/reent.h:17,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdio.h:60,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/cores/esp32/Arduino.h:27,
                 from sketch/e01_Hello_ESP32_test.ino.cpp:1:
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:78:3: note: candidate: 'constexpr long double std::abs(long double)'
   abs(long double __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:74:3: note: candidate: 'constexpr float std::abs(float)'
   abs(float __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:70:3: note: candidate: 'constexpr double std::abs(double)'
   abs(double __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:61:3: note: candidate: 'long long int std::abs(long long int)'
   abs(long long __x) { return __builtin_llabs (__x); }
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:56:3: note: candidate: 'long int std::abs(long int)'
   abs(long __i) { return __builtin_labs(__i); }
   ^~~
In file included from /Users/user/Documents/Arduino/libraries/souliss/Souliss.h:222,
                 from /Users/user/Documents/e01_Hello_ESP32_test/e01_Hello_ESP32_test.ino:30:
/Users/user/Documents/Arduino/libraries/souliss/base/LocalIO.cpp: In function 'U8 Souliss_DigInHoldSteps_Helper(U8, U8, U8*, U8, U8, U16)':
/Users/user/Documents/Arduino/libraries/souliss/base/LocalIO.cpp:679:61: error: call of overloaded 'abs(long unsigned int)' is ambiguous
  else if( (InPin[pin]==PINSET) && (abs(millis()-souliss_time) < step_duration) )
                                                             ^
In file included from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/cstdlib:75,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/stdlib.h:36,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/assert.h:21,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/sys/reent.h:503,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/sys/reent.h:17,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdio.h:60,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/cores/esp32/Arduino.h:27,
                 from sketch/e01_Hello_ESP32_test.ino.cpp:1:
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdlib.h:74:5: note: candidate: 'int abs(int)'
 int abs (int);
     ^~~
In file included from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/cstdlib:77,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/stdlib.h:36,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/assert.h:21,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/sys/reent.h:503,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/sys/reent.h:17,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdio.h:60,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/cores/esp32/Arduino.h:27,
                 from sketch/e01_Hello_ESP32_test.ino.cpp:1:
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:78:3: note: candidate: 'constexpr long double std::abs(long double)'
   abs(long double __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:74:3: note: candidate: 'constexpr float std::abs(float)'
   abs(float __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:70:3: note: candidate: 'constexpr double std::abs(double)'
   abs(double __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:61:3: note: candidate: 'long long int std::abs(long long int)'
   abs(long long __x) { return __builtin_llabs (__x); }
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:56:3: note: candidate: 'long int std::abs(long int)'
   abs(long __i) { return __builtin_labs(__i); }
   ^~~
In file included from /Users/user/Documents/Arduino/libraries/souliss/Souliss.h:222,
                 from /Users/user/Documents/e01_Hello_ESP32_test/e01_Hello_ESP32_test.ino:30:
/Users/user/Documents/Arduino/libraries/souliss/base/LocalIO.cpp:684:61: error: call of overloaded 'abs(long unsigned int)' is ambiguous
  else if( (InPin[pin]==PINSET) && (abs(millis()-souliss_time) > step_duration) )
                                                             ^
In file included from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/cstdlib:75,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/stdlib.h:36,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/assert.h:21,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/sys/reent.h:503,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/sys/reent.h:17,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdio.h:60,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/cores/esp32/Arduino.h:27,
                 from sketch/e01_Hello_ESP32_test.ino.cpp:1:
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdlib.h:74:5: note: candidate: 'int abs(int)'
 int abs (int);
     ^~~
In file included from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/cstdlib:77,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/stdlib.h:36,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/assert.h:21,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/sys/reent.h:503,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/sys/reent.h:17,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdio.h:60,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/cores/esp32/Arduino.h:27,
                 from sketch/e01_Hello_ESP32_test.ino.cpp:1:
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:78:3: note: candidate: 'constexpr long double std::abs(long double)'
   abs(long double __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:74:3: note: candidate: 'constexpr float std::abs(float)'
   abs(float __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:70:3: note: candidate: 'constexpr double std::abs(double)'
   abs(double __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:61:3: note: candidate: 'long long int std::abs(long long int)'
   abs(long long __x) { return __builtin_llabs (__x); }
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:56:3: note: candidate: 'long int std::abs(long int)'
   abs(long __i) { return __builtin_labs(__i); }
   ^~~
In file included from /Users/user/Documents/Arduino/libraries/souliss/Souliss.h:222,
                 from /Users/user/Documents/e01_Hello_ESP32_test/e01_Hello_ESP32_test.ino:30:
/Users/user/Documents/Arduino/libraries/souliss/base/LocalIO.cpp:690:87: error: call of overloaded 'abs(long unsigned int)' is ambiguous
  else if( (InPin[pin]==PINACTIVE || InPin[pin]==PINUSED) && (abs(millis()-souliss_time) > step_duration) )
                                                                                       ^
In file included from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/cstdlib:75,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/stdlib.h:36,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/assert.h:21,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/sys/reent.h:503,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/sys/reent.h:17,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdio.h:60,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/cores/esp32/Arduino.h:27,
                 from sketch/e01_Hello_ESP32_test.ino.cpp:1:
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdlib.h:74:5: note: candidate: 'int abs(int)'
 int abs (int);
     ^~~
In file included from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/cstdlib:77,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/stdlib.h:36,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/assert.h:21,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/sys/reent.h:503,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/sys/reent.h:17,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdio.h:60,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/cores/esp32/Arduino.h:27,
                 from sketch/e01_Hello_ESP32_test.ino.cpp:1:
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:78:3: note: candidate: 'constexpr long double std::abs(long double)'
   abs(long double __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:74:3: note: candidate: 'constexpr float std::abs(float)'
   abs(float __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:70:3: note: candidate: 'constexpr double std::abs(double)'
   abs(double __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:61:3: note: candidate: 'long long int std::abs(long long int)'
   abs(long long __x) { return __builtin_llabs (__x); }
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:56:3: note: candidate: 'long int std::abs(long int)'
   abs(long __i) { return __builtin_labs(__i); }
   ^~~
In file included from /Users/user/Documents/Arduino/libraries/souliss/Souliss.h:222,
                 from /Users/user/Documents/e01_Hello_ESP32_test/e01_Hello_ESP32_test.ino:30:
/Users/user/Documents/Arduino/libraries/souliss/base/LocalIO.cpp:694:61: error: call of overloaded 'abs(long unsigned int)' is ambiguous
   U8 powered_lights_count = (U8) ( abs(millis()-souliss_time) / step_duration );
                                                             ^
In file included from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/cstdlib:75,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/stdlib.h:36,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/assert.h:21,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/sys/reent.h:503,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/sys/reent.h:17,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdio.h:60,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/cores/esp32/Arduino.h:27,
                 from sketch/e01_Hello_ESP32_test.ino.cpp:1:
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdlib.h:74:5: note: candidate: 'int abs(int)'
 int abs (int);
     ^~~
In file included from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/cstdlib:77,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/stdlib.h:36,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/assert.h:21,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/sys/reent.h:503,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/tools/sdk/esp32/include/newlib/platform_include/sys/reent.h:17,
                 from /Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/sys-include/stdio.h:60,
                 from /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/cores/esp32/Arduino.h:27,
                 from sketch/e01_Hello_ESP32_test.ino.cpp:1:
Multiple libraries were found for "WiFi.h"
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:78:3: note: candidate: 'constexpr long double std::abs(long double)'
   abs(long double __x)
   ^~~
 Used: /Users/user/Library/Arduino15/packages/esp32/hardware/esp32/2.0.3/libraries/WiFi
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:74:3: note: candidate: 'constexpr float std::abs(float)'
   abs(float __x)
   ^~~
 Not used: /Applications/Arduino.app/Contents/Java/libraries/WiFi
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:70:3: note: candidate: 'constexpr double std::abs(double)'
   abs(double __x)
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:61:3: note: candidate: 'long long int std::abs(long long int)'
   abs(long long __x) { return __builtin_llabs (__x); }
   ^~~
/Users/user/Library/Arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/gcc8_4_0-esp-2021r2-patch3/xtensa-esp32-elf/include/c++/8.4.0/bits/std_abs.h:56:3: note: candidate: 'long int std::abs(long int)'
   abs(long __i) { return __builtin_labs(__i); }
   ^~~
In file included from /Users/user/Documents/Arduino/libraries/souliss/frame/vNet/vNet.cpp:34,
                 from /Users/user/Documents/Arduino/libraries/souliss/Souliss.h:182,
                 from /Users/user/Documents/e01_Hello_ESP32_test/e01_Hello_ESP32_test.ino:30:
/Users/user/Documents/Arduino/libraries/souliss/frame/vNet/tools/oFrame.c: In function 'U8 oFrame_Available()':
/Users/user/Documents/Arduino/libraries/souliss/frame/vNet/tools/oFrame.c:126:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
exit status 1
Error compiling for board WEMOS D1 MINI ESP32.
```

--- a/base/LocalIO.cpp
+++ b/base/LocalIO.cpp
@@ -296,7 +296,7 @@ inline U8 Souliss_DigInHold_Helper(U8 pin, U8 pin_value, U8 value, U8 value_hold
 
 		return MaCaco_NODATACHANGED;
 	}
-	else if(pin_value && (abs(millis()-souliss_time) > holdtime) && (InPin[pin]==PINSET))
+	else if(pin_value && (abs((long)(millis()-souliss_time)) > holdtime) && (InPin[pin]==PINSET))
 	{
 		InPin[pin] = PINACTIVE;								// Stay there till pushbutton is released
 
@@ -340,7 +340,7 @@ inline U8 Souliss_DigInHold2_Helper(U8 pin, U8 pin_value, U8 value, U8 value_hol
 
 		return MaCaco_NODATACHANGED;
 	}
-	else if(pin_value && (abs(millis()-souliss_time) > holdtime) && (InPin[pin]==PINSET))
+	else if(pin_value && (abs((long)(millis()-souliss_time)) > holdtime) && (InPin[pin]==PINSET))
 	{
 		InPin[pin] = PINACTIVE;								// Stay there till pushbutton is released
 
@@ -395,7 +395,7 @@ inline U8 Souliss_DigKeepHold_Helper(U8 pin, U8 pin_value, U8 value, U8 value_ho
 
 		return MaCaco_NODATACHANGED;
 	}
-	else if(pin_value && (abs(millis()-souliss_time) > holdtime) && ((InPin[pin]==PINSET) || (InPin[pin]==PINACTIVE)))
+	else if(pin_value && (abs((long)(millis()-souliss_time)) > holdtime) && ((InPin[pin]==PINSET) || (InPin[pin]==PINACTIVE)))
 	{
 		souliss_time = millis();
 		InPin[pin] = PINACTIVE;								// Stay there till pushbutton is released
@@ -676,22 +676,22 @@ inline U8 Souliss_DigInHoldSteps_Helper(U8 pin, U8 pin_value, U8 *memory_map, U8
 		// short presses are handled on button release event
 		return MaCaco_NODATACHANGED;
 	}
-	else if( (InPin[pin]==PINSET) && (abs(millis()-souliss_time) < step_duration) )
+	else if( (InPin[pin]==PINSET) && (abs((long)(millis()-souliss_time)) < step_duration) )
 	{
 		// still short press -> do nothing
 		return MaCaco_NODATACHANGED;
 	}
-	else if( (InPin[pin]==PINSET) && (abs(millis()-souliss_time) > step_duration) )
+	else if( (InPin[pin]==PINSET) && (abs((long)(millis()-souliss_time)) > step_duration) )
 	{
 		// long press -> do nothing now, but remember it with InPin value
 		InPin[pin]=PINACTIVE;
 		return MaCaco_NODATACHANGED;
 	}
-	else if( (InPin[pin]==PINACTIVE || InPin[pin]==PINUSED) && (abs(millis()-souliss_time) > step_duration) )
+	else if( (InPin[pin]==PINACTIVE || InPin[pin]==PINUSED) && (abs((long)(millis()-souliss_time)) > step_duration) )
 	{
 		// this cycle is executed while the button is kept pressed
 		// the current input is 1, the previous input was 1 and some time passed from the first press
-		U8 powered_lights_count = (U8) ( abs(millis()-souliss_time) / step_duration );
+		U8 powered_lights_count = (U8) ( abs((long)(millis()-souliss_time)) / step_duration );
 
 		if( InPin[pin]==PINACTIVE )
 		{

--- a/frame/vNet/tools/oFrame.c
+++ b/frame/vNet/tools/oFrame.c
@@ -118,8 +118,7 @@ U8 oFrame_isBusy()
 U8 oFrame_Available()
 {
 	// If a frame is defined
-	if(actualframe)
-		if (oFrame_GetLenght())
+	if(actualframe && oFrame_GetLenght())
 			return 1;
 	else
 		return 0;

